src/java/test/org/apache/zookeeper/test/NettyLocalSuiteTest.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zookeeper.test;
+
+import org.junit.runners.Suite;
+
+/**
+ * Run tests with: Netty Client against Netty server
+ */
+@Suite.SuiteClasses({
```

Comment:
@afine Thank you for your review.
I have addressed most of your comments (fix imports, duplicate code...)

Regarding the tests, 
It will be interesting in order to see that all the code can run cleanly without network.
But IMHO a new system property/run time flag will add the need to run always twice the full suite at every build/release.
I have added the least possible testcases to cover basic features so that future refactor will not drop the ability to use the Local Transport.
At this moment I think it is not the time to add an official "local transport" netty connection factory.


Regarding server-to-server communications:
I can't find how to enable Netty for server-to-server communications, I will dig deeply into the code. It seems that the Learner class uses directly a Socket, so it is not possible to leverage Netty local transport feature.